### PR TITLE
Decimal256 coercion

### LINKED ISSUE #7033: DataFusion-Expr TypeCoercion missing Decimal256
### Describe the bug

Currently in datafusion_expr TypeCoercion `DataType::Decimal256(_, _)` is missing from 2 key functions.

1. `is_signed_numeric`
2. `is_decimal`

Currently only `Decimal128(_, _)` is considered. This seems to be a simple oversight where it was not added when `Decimal256` support was added.

### To Reproduce

_No response_

### Expected behavior

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jdye64**
# Which issue does this PR close?

Closes #7033

### CODE REVIEW

**Review by andygrove — APPROVED** [review]
LGTM. Thanks @jdye64

File: datafusion/expr/src/type_coercion/aggregates.rs [hunk 1]
```diff
@@ -480,7 +480,7 @@ pub fn is_avg_support_arg_type(arg_type: &DataType) -> bool {
         _ => matches!(
             arg_type,
             arg_type if NUMERICS.contains(arg_type)
-                || matches!(arg_type, DataType::Decimal128(_, _))
+                || matches!(arg_type, DataType::Decimal128(_, _)| DataType::Decimal256(_, _))
```
Comment (alamb) [inline datafusion/expr/src/type_coercion/aggregates.rs:483]: Doesn't this change imply that people can run `AVG(dec256_col)` now? I tried to run this query


```sql
create table t as values (arrow_cast(123, 'Decimal256'));
select AVG(column1) from t;
```

But it seems like datafusion support for Decimal256 is still in progress

```
❯ create table t as values (arrow_cast(123, 'Decimal256(10,3)'));
Optimizer rule 'simplify_expressions' failed
caused by
This feature is not implemented: Can't create a scalar from array of type "Decimal256(10, 3)"
```

**Review by alamb — APPROVED** [review]
Thank you @jdye64 

This seems like a fine incremental improvement to me, but it seems like there is a lot more work needed to properly support Decimal 256, just FYI

**Comment by jdye64** [comment]
I agree. I stumbled across this in a dask-sql test and while this solves my immediate issue there is more to it for sure

**Review by Dandandan — APPROVED** [review]

File: datafusion/expr/src/type_coercion/aggregates.rs (hunk 1 above)
Comment (jdye64) [reply datafusion/expr/src/type_coercion/aggregates.rs:483]: hmm yes, I was under the impression that it should. In Dask-Sql's case we are not using the physical planner from DataFusion and rather using our own so maybe this error is coming in from some similar missed `Decimal256` coverage there? That would also explain why I don't see this issue when running as a library with dask-sql I guess?

Comment (Dandandan) [reply datafusion/expr/src/type_coercion/aggregates.rs:483]: `simplify_expressions` is an optimization pass on the logical plan.
I suppose you're not using those or having your own.

Anyway, would be good to have some more support / (end-to-end) tests for `Decimal256`.

**Pull request merged**